nncf/quantization/algorithms/accuracy_control/algorithm.py

```diff
@@ -0,0 +1,342 @@
+
+            if not ranked_groups:
+                nncf_logger.info(
+                        'All layers have been checked and the AccuracyAwareQuantization '
+                        'will not be able to achieve the required accuracy drop')
+                removed_all = True
+                break
+
+            # greedy removal of the FQ node with the highest importance score
+            current_group = ranked_groups.pop()
+            current_model = revert_operations_to_floating_point_precision(
+                current_group.operations,
+                current_group.quantizers,
+                previous_model,
+                quantized_model_graph,
+                algo_backend.create_command_to_remove_quantizer,
+                algo_backend.create_command_to_update_bias,
+                algo_backend.create_command_to_update_weight)
+
+            nncf_logger.debug(f'Removed a block of {len(current_group.quantizers)} quantizers:'
+                              f'\n{_create_message(current_group.quantizers)}')
+            nncf_logger.info(f'Reverted {len(current_group.operations)} operations to the {precision_change_to} '
+                             f'precision: \n{_create_message(current_group.operations)}')
+
+            current_num_quantizers = current_num_quantizers - len(current_group.quantizers)
+            all_removed_nodes.extend(current_group.quantizers)
+            all_reverted_ops.update(current_group.operations)
+
+            # Calculate drop for new quantization scope.
+            current_metric = validation_fn(algo_backend.prepare_for_inference(current_model),
+                                           validation_dataset.get_data())
+            current_accuracy_drop = initial_metric - current_metric
+            nncf_logger.info('Accuracy drop with the new quantization scope is %s', float(current_accuracy_drop))
+
+            if current_num_quantizers == 0:
+                nncf_logger.info('All quantizers were removed from the model.')
+                removed_all = True
+                break
+
+            # Accuracy was restored to the acceptable drop.
+            if current_accuracy_drop <= max_drop:
+                reached_required_drop = True
+                break
+
+            # Continue greedy quantizer remove
+            if max_drop < current_accuracy_drop <= previous_accuracy_drop \
+                    or (current_accuracy_drop > previous_accuracy_drop and is_step_back):
+                is_step_back = False
+                previous_accuracy_drop = current_accuracy_drop
+                continue
+
+            if current_accuracy_drop > previous_accuracy_drop and USE_PREVIOUS_IF_DROP_INCREASE:
+                current_model = previous_model
+                all_removed_nodes = all_removed_nodes[:len(all_removed_nodes)-len(current_group.quantizers)]
+                all_reverted_ops.difference_update(current_group.operations)
+                if exclude_bad_nodes:
```

Comment:
How can we reach it if `exclude_bad_nodes` is always `False`?

Reply:
This parameter was removed.